crates/ruff/src/rules/flake8_pyi/rules/unused_private_type_declaration.rs

```diff
@@ -0,0 +1,62 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_semantic::Binding;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for the presence of unused private `typing.Protocol` definitions.
+///
+/// ## Why is this bad?
+/// A private `typing.Protocol` that is defined but not used is likely a mistake, and should
+/// be removed to avoid confusion.
```

Comment:
Should we suggest that it could be made public as an alternative fix?

Reply:
That sounds good to me